### PR TITLE
Initial basic Org markup support thanks to tree-sitter-org

### PR DESCRIPTION
**Pull request by zetashift**
Using (and many thanks to):
https://github.com/milisims/tree-sitter-org
https://github.com/nvim-orgmode/orgmode/tree/master/queries/org

And the-mikedavis from the matrix chat. I think I need to add a few more things like docs, but I would like to first know and what I can add and what I can improve? 
Ideally I'd want the headers to be a different colors each(I think this is a theming thing though) and property drawers to be colored different as well. Not entirely sure how to go about it?

### CODE REVIEW

**Comment by the-mikedavis** [comment]
This is looking good! For the docs, you'll need to run `cargo xtask docgen` and commit the changes. These queries will allow the headers to be highlighted differently but whether they'll be different colors is up to whichever theme you use. [Here](https://github.com/helix-editor/helix/blob/cfd992b1511655fc2116af4ddd1cd0dbb3bcb85e/runtime/themes/rose_pine.toml#L52-L57)'s a theme that has different colors set up.

**Comment by zetashift** [comment]
Alright pretty happy with this, I kind of want to have different highlights for TODO and DONE etc, but a tree-sitter parser shouldn't be responsible for that.

Also wanted a different query for links, but I couldn't find the query for that.

image

An example org file: https://writequit.org/denver-emacs/presentations/files/example.org.html

**Comment by the-mikedavis** [comment]
Giving this a try locally, I see that some markup is more difficult to highlight than I expected.

```org
/foo bar baz/
```

won't match the

```scm
((expr) @markup.italic
 (#match? @markup.italic "/.*/"))
```

pattern because it gets parsed as three separate `expr` nodes. We could highlight a case like this with

```scm
(((expr) @markup.italic @_start
  (expr)* @markup.italic
  (expr) @markup.italic @_end)
 (#match? @_start "^/")
 (#match? @_end "/$"))
```

but I tried that locally and it degrades highlighting performance really bad. I think for now these queries are the best we can do. We could try out another parser, but milisims' seems like the best option of the [available parsers](https://github.com/search?q=tree-sitter-org). Ideally this parser should be refactored so it would parse markup into named nodes instead of leaving it to the queries. (I bet that this would be a pretty large refactor of the grammar though.)

File: runtime/queries/org/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  (#match? @markup.bold "\\*.*\\*"))
+ 
+ ((expr) @markup.italic
+  (#match? @markup.italic "/.*/"))
+
+ ((expr) @markup.raw.inline
+  (#match? @markup.raw.inline "~.*~"))
+ 
+ ((expr) @markup.quote
+  (#match? @markup.quote "=.*="))
```
Comment (the-mikedavis) [inline runtime/queries/org/highlights.scm:24]:
```suggestion
(block) @markup.raw.block
(list) @markup.list.unnumbered
(directive) @markup.label
(property_drawer) @markup.label
 

((expr) @markup.bold
 (#match? @markup.bold "\\*.*\\*"))
 
((expr) @markup.italic
 (#match? @markup.italic "/.*/"))

((expr) @markup.raw.inline
 (#match? @markup.raw.inline "~.*~"))
 
((expr) @markup.quote
 (#match? @markup.quote "=.*="))

```

looks like some extra spaces snuck in here, and there's a missing trailing newline. Would you mind adding one to the end of `injections.scm` as well?

**Review by the-mikedavis — COMMENTED** [review]
looking good, just some nits about the formatting 👍

File: runtime/queries/org/highlights.scm (hunk 1 above)
Comment (zetashift) [reply runtime/queries/org/highlights.scm]: Did I ever tell you I suck at Git? Cause I do, I think now the spaces are gone (atleast I think so), and newlines are added, let me know if anything else needs fixing.

**Review by the-mikedavis — APPROVED** [review]
Nice work!

**Pull request merged**